test/unit/cli/test_setup_s3_bucket.py

```diff
@@ -0,0 +1,31 @@
+import pytest
+from test.unit.cli import CliRunner
+from exasol.ds.sandbox.cli.commands.setup_s3_bucket import setup_s3_bucket
+from unittest.mock import patch
+from exasol.ds.sandbox.lib.cloudformation_templates import (
+    VmBucketCfTemplate,
+    ExampleDataCfTemplate,
+)
+
+
+@pytest.fixture
+def cli():
+    return CliRunner(setup_s3_bucket)
+
+
+def test_no_purpose(cli):
+    assert cli.run().failed("Missing option '--purpose'")
+
+
+def test_vm_bucket(cli):
+    with patch.object(VmBucketCfTemplate, 'setup') as setup:
+        cli.run("--purpose", "vm")
+    assert cli.succeeded()
+    setup.assert_called_once()
+
+
+def test_example_data_bucket(cli):
+    with patch.object(ExampleDataCfTemplate, 'setup') as setup:
+        cli.run("--purpose", "example-data")
+    assert cli.succeeded()
+    setup.assert_called_once()
```

Comment:
using `assert_called_* ` with pytest is usually not the best thing. I usually suggest, `assert setup.mock_calls = [call()]`

Reply:
See next push